packages/signature-controller/src/SignatureController.ts

```diff
@@ -161,6 +163,11 @@ export type SignatureControllerOptions = {
    */
   decodingApiUrl?: string;
 
+  /**
+   * Function to get features flag information
+   */
+  getFeatureFlags?: () => FeatureFlags;
```

Comment:
Rather than coupling multiple features to a single callback, could we have a single `isDecodingEnabled` callback as we do in other controllers.

So a client can have alternate logic per feature and not have the logic overlap?

Reply:
I found that other controllers like snap-controller are using similar approach
```
getFeatureFlags: () => {
        return {
          disableSnaps:
            this.preferencesController.state.useExternalServices === false,
        };
      },
```

Reply:
It's not standard in the `TransactionController` which is our closest parallel, plus it couples unrelated feature flags together which we want to avoid.

We don't want a client to have to group multiple feature enablement in a single callback when the logic to enable them could be entirely different, nor do we want to insist a client specify multiple feature flags at once as each feature my have different default enablement.

Reply:
Done